### PR TITLE
fix: Show starred docs in sidebar

### LINKED ISSUE #1313: Show starred docs in sidebar
Currently the treatment for starred is the same as other docs, they basically just get added to another screen that acts the same as any other collection . While this is _okay_ it's barely better than exposing the database schema to the user – starred docs should be **easier to access** than other docs.

### Proposed changes: 

- Remove the existing "Starred" item at the top of the sidebar
- Add a "Starred" heading to the sidebar when the user has starred documents above collections
- Render upto X (10?) documents below it in a flat list and a "Show more…" link at the end
- The "show more" link should navigate to the existing starred page for when there are many starred items to filter through (Use similar styling as the e.g. "New collection")

### PR DESCRIPTION
**Pull request by iamsaumya**
Closes #1313 

- [x] Change starred to bookmarked on user-facing instances. 
- [x] Remove starred scene

### CODE REVIEW

**Comment by iamsaumya** [comment]
@tommoor, please take a look.

**Comment by tommoor** [comment]
Thank you – leave this with me for a few, having seen it in action might need to rethink a little 😆

**Comment by iamsaumya** [comment]
Haha, Sure 👍🏻

**Comment by tommoor** [comment]
Okay, here's what I think is needed to make this workable – I'd be interested in your feedback…

- "Show more…" should only appear when user has an exceptionally large amount of starred items, let's say >10.
- Given the above the `/starred` route becomes quite redundant and rarely used, I'm thinking we can convert this to a filter on the search so you can filter to starred docs. This could be tackled in a separate upfront PR.
- Repeating the ⭐️ icon in the sidebar is a bit much, It might be best to leave this space empty, or show a simple bullet point (will need to do some design explorations here).
- When a starred doc is clicked it should be highlighted in the starred section too
- If a starred doc has child documents it should be possible to expand within the starred section

**Comment by iamsaumya** [comment]
> Given the above, the /starred route becomes quite redundant and rarely used. I'm thinking we can convert this to a filter on the search so you can filter to starred docs. This could be tackled in a separate upfront PR.

Given how the search works, we need to pass some queries to it with the filters. It will be weird to show docs without any queries. 

>Repeating the star icon in the sidebar is a bit much, It might be best to leave this space empty, or show a simple bullet point (will need to do some design explorations here).

A suggestion - Given that we will only show at most 10 docs, Is it possible to have some sort of color binding for each star? It will look lively in the sidebar.

**Comment by tommoor** [comment]
> Is it possible to have some sort of color binding for each star?

I'm having trouble imagining this working well, if the colors are "random" then they have no meaning and if they're user choosable then that's a lot to ask of folks

**Comment by iamsaumya** [comment]
Random in a sense, we will have a list of 10 colors, the first starred doc will always have that specific color, and so on

Or maybe we can pick the color from their collection's icon?

**Comment by iamsaumya** [comment]
So I guess we will go without any icons. 

Any comments for the suggestions on the approach to the `search` page?

**Comment by tommoor** [comment]
> Any comments for the suggestion on the approach to the search page?

I think it could make sense to add a "Starred documents" to this filter and expose it via a query string parameter, thoughts?

image

**Comment by iamsaumya** [comment]
Adding a filter for `starred documents` is a good idea indeed. 

So when a user clicks on `show more...`, should they be redirected to the search page with the filter selected? 

If so, I am hesitant about the idea to show the `starred documents` without any `search query` on the search page as it feels like an anti-pattern.

**Comment by tommoor** [comment]
Fair, I suppose I didn't think that one through thoroughly enough 😆 

In that case perhaps "Show more" just expands the display in the sidebar to load + show all of the users starred items

**Comment by iamsaumya** [comment]
Sounds great! 🤜🏻🤛🏻

**Comment by iamsaumya** [comment]
I made some progress here.

Screenshot from 2021-07-23 17-27-45

**Comment by iamsaumya** [comment]
I will complete the above todos when we are happy with this implementation.

File: app/components/Sidebar/components/SidebarLink.js [hunk 1]
```diff
@@ -156,20 +157,18 @@ const Link = styled(NavLink)`
...
+  }
 
-    &:hover {
-      color: ${(props) =>
-        props.$isActiveDrop ? props.theme.white : props.theme.text};
-    }
+  &:hover {
+    color: ${(props) =>
+      props.$isActiveDrop ? props.theme.white : props.theme.text};
   }
```
Comment (iamsaumya) [inline app/components/Sidebar/components/SidebarLink.js:172]: Apparently, we had an extra curly bracket here.

Comment (tommoor) [reply app/components/Sidebar/components/SidebarLink.js:172]: I think you just accidentally moved it above?

Comment (iamsaumya) [reply app/components/Sidebar/components/SidebarLink.js:172]: Oh, so `&:hover` selection is inside the media query?

**Comment by iamsaumya** [comment]
@tommoor, I think this is done. Please give it a review 😄

**Comment by iamsaumya** [comment]
Hey, just bumping up to your notifications, in case you have forget about this 😅

**Comment by tommoor** [comment]
Haven't forgotten – I think the functionality is basically spot-on, just pondering on how these are aligned/displayed still, there are three ways to align these items and each one looks out of alignment in a different way depending on if there is a disclosure or not 😢 

thinking through that…

**Comment by iamsaumya** [comment]
I vouch for this.

I tried different designs and chose this implementation in the end, which looked best to me. 

Still, I feel that something isn't right about the presentation.

**Comment by tommoor** [comment]
The problem as I'm sure you saw is that documents at the same level in hierarchy are not aligned horizontally because of the disclosure, but if you _do_ align them it leaves a gap that looks strange, for example:

Screen Shot 2021-07-26 at 3 16 42 PM

You could also push the disclosure into the gutter, but this looks wrong because the titles kind of _want_ to be aligned with all the other text:

Screen Shot 2021-07-26 at 3 17 50 PM

**Comment by iamsaumya** [comment]
Yupp, that's why I aligned everything in a line, it looked the best of all 😅

**Comment by tommoor** [comment]
telegram-cloud-photo-size-1-4904638037267425657-y

Possible way forward, discussed in chat.

**Comment by iamsaumya** [comment]
LGTM 👍🏻

**Comment by iamsaumya** [comment]
Do we want the bookmark section to be collapsible?

**Comment by tommoor** [comment]
Would be nice – show a disclosure on hover of heading?

**Comment by iamsaumya** [comment]
And when it is closed, should the disclosure be visible?

**Comment by tommoor** [comment]
Yes!

**Comment by iamsaumya** [comment]
Done ✅

**Comment by tommoor** [comment]
We spoke privately on this and this design ended up being the front-runner 😅 

telegram-cloud-photo-size-1-4907198000984598886-y

**Comment by iamsaumya** [comment]
Please review 😃

**Review by tommoor — CHANGES_REQUESTED** [review]
Thanks! Some feedback – upon opening the branch my sidebar looks like this?

image

- [x] Everything should be expanded by default
- [x] The headings should use the same style as other items (see design above), perhaps we could make the entire thing clickable?
- [x] Lets keep the disclosure visible at all times with this design
- [x] "New collection…" item should be inset the same amount as collections

**Comment by iamsaumya** [comment]
`depth` prop is one of the best implementations I have ever come across. No need to worry about `px`. Just pass an appropriate number 😍

**Comment by tommoor** [comment]
Sorry, found a few more things – but after playing with it I'm confident this layout will work 🙏 

- [ ] If there are no starred docs then hide the heading, currently looks like:
      image

- [ ]  The top "drop cursor" in the collections section is positioned too high up, try dragging a collection to the top to see. This also makes the bottom half of the "Collections" header unclickable.
- [ ] Is it possible to inset the loading state under "starred" to match the content?
- [ ] Is it possible to include the "..." menu on starred items in the sidebar the same as in the regular collection?

**Comment by iamsaumya** [comment]
> If there are no starred docs then hide the heading, currently looks like

Sure. Actually, I deliberately took the decision to show it. I thought users should be aware that this estate is where starred documents will appear in future.

**Comment by tommoor** [comment]
>  I deliberately took the decision to show it. I thought users should be aware that this estate is where starred documents will appear

I can see the logic, but it's kind of 100px of prime real estate pushing down the main content :) Thankfully we have lots of ways to star a doc so it should be discoverable, it can also be in a future onboarding if needed.

File: app/components/Sidebar/components/StarredLink.js [hunk 2]
```diff
@@ -88,6 +88,7 @@ function StarredLink({ depth, title, to, documentId, collectionId }: Props) {
                 document={document}
                 onOpen={handleMenuOpen}
                 onClose={handleMenuClose}
+                modal={false}
```
Comment (iamsaumya) [inline app/components/Sidebar/components/StarredLink.js:91]: Sometimes the menu is placed in the top left corner. This seems to remove it.

**Comment by iamsaumya** [comment]
Ready for review 😄

**Comment by tommoor** [comment]
Did a fair few final bits of polish here, lmk if anything stands out as problematic – otherwise I think this should be good to merge!

**Comment by iamsaumya** [comment]
Thanks for the help here 😄 . Looks good to me!

**Review by tommoor — APPROVED** [review]
🙏  I do hope that this change is well received

**Comment by iamsaumya** [comment]
It will be fun to have feedback from someone who has starred 100s of docs 😁

**Pull request merged**